documentation/public/user-guide/07-semantic-analysis/01-compilation-units/index.md

```diff
@@ -0,0 +1,60 @@
+# 7.1. Compilation Units
+
+In order for Slang to perform semantic analysis on a program, it first needs to compile a list of all source files into a single `CompilationUnit`.
+
+We offer a `CompilationBuilder` API that allows you easily do that.
+It will parse the provided source files, analyze the import statements within, and then transitively resolve all dependencies, until all source files have been loaded.
+
+It is important to note that Slang is designed to be modular, and work in any environment (CLI, browser, etc..).
+This means it cannot access the file system directly, or make any assumptions about where your contracts are, and where their dependencies are installed.
+
+Instead, it expects the user to provide it with a couple of callbacks to handle these tasks:
```

Comment:
```suggestion
To address these limitations, it expects the user to provide a couple of callbacks to handle these tasks:
```

Reply:
I don't think that is a limitation per se, rather it is a feature/by design.

Reply:
Good point!